django/test/html.py

```diff
@@ -95,6 +95,9 @@ def _count(self, element, count=True):
         if not isinstance(element, six.string_types):
             if self == element:
                 return 1
+        if isinstance(element, RootElement):
+            raise ValueError('Needle HTML does not have a root element')
```

Comment:
Should we add some context to the message (like part of the HTML snippet) to help identify the needle? I'm a bit worried developers will have a tough time figuring out what the message means.